packages/common-frontend/src/features/engine/slice.ts

```diff
@@ -161,12 +164,13 @@ export const engineSlice = createSlice({
   reducers: {
     setFromInit: (state, action: PayloadAction<DEngineInitPayload>) => {
       const { notes, wsRoot, schemas, vaults, config } = action.payload;
+      NOTE_FNAMES_DICT = new NoteFNamesDict(_.values(notes));
       state.notes = notes;
       state.wsRoot = wsRoot;
       state.schemas = schemas;
       state.vaults = vaults;
       state.config = config;
-      state.noteFName = new NoteFNamesDict(_.values(notes));
+      state.noteFName = NOTE_FNAMES_DICT;
```

Comment:
I think we'll run into that similar issue I encountered w/ the GraphView optimization where some code complains about redux state not being serializable.  We should create a more primitive type for this dict like `[key: string]: string;` and use that type here (and in the graph view) instead.

Consequently, we'll also need to amend the arguments of `addOrUpdateParents`

Reply:
yeah, this is why we added NOTE_FNAMES_DICT in the module instead of inside of redux. @tma66 do we still need this in state?

Reply:
Looks like its being used in graph view. https://github.com/dendronhq/dendron/pull/2903. I'd have to expose  `NOTE_FNAMES_DICT` if we wanted to remove from state. @jonathanyeung What do you think here?

Reply:
As discussed offline, since this is a pre-existing issue, let's defer that to the next iteration.